Branch-SDK/src/io/branch/referral/AnimatedDialog.java

```diff
@@ -0,0 +1,148 @@
+package io.branch.referral;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+import android.view.Gravity;
+import android.view.KeyEvent;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.view.WindowManager;
+import android.view.animation.AccelerateInterpolator;
+import android.view.animation.Animation;
+import android.view.animation.DecelerateInterpolator;
+import android.view.animation.TranslateAnimation;
+
+/**
+ * <p>Class for creating a Dialog which open and closes with an animation to the content view </p>
+ */
+class AnimatedDialog extends Dialog {
+    private static boolean isClosing_ = false;
+    Context context_;
+    public AnimatedDialog(Context context) {
+        super(context);
+        init(context);
+    }
+
+    public AnimatedDialog(Context context, int theme) {
+        super(context, theme);
+        init(context);
+    }
+
+    public AnimatedDialog(Context context, boolean cancelable, OnCancelListener cancelListener) {
+        super(context, cancelable, cancelListener);
+        init(context);
+    }
+
+    //--------------------- Public  methods -------------//
+
+    /**
+     * <p> Opens the dialog with an animation to the content View.</p>
+     */
+    @Override
+    public void show() {
+        slideOpen();
+    }
+
+    /**
+     * <p> Cancels the dialog with an animation to the content View.</p>
+     */
+    @Override
+    public void cancel() {
+        slideClose();
+    }
+
+    /**
+     * </p> Dismiss the dialog.</p>
+     */
+    @Override
+    public void dismiss() {
```

Comment:
If you're just calling the super here, probably don't need to override this method.